### DISCUSSION TITLE
Subscription switching nodes several times

### DISCUSSION THREAD

**Discussion by logankp** [ravendb/ravendb#13018] [Q&A]
Using RavenDb 4.2 cloud, 3 node cluster.

We're seeing issues occasionally with subscriptions where they seem to be switching nodes several times an hour. For example, we've seen the following exception over 100 times in 30+ hours just recently, and several more per hour about a month ago: `Subscription With Id 'ActivityForMoldDetailSubscription' cannot be processed by current node 'B', it will be redirected to ]`. The subscription seems to stop processing altogether and will not re-start until we fully re-startour application. I thought the subscription was supposed to re-connect automatically? Any thoughts on why it keeps firing the `OnSubscriptionConnectionRetry` event until the app gets restarted?

**Comment by garayx** [comment]
Hi 

This exception is often an indication to unstable cluster or unreachable database, in such case it will try reconnect and will fire the `OnSubscriptionConnectionRetry` event. 

Can you check the cluster status, and the database group when the issue happens ? 
[https://ravendb.net/docs/article-page/5.2/csharp/studio/database/settings/manage-database-group](https://ravendb.net/docs/article-page/5.2/csharp/studio/database/settings/manage-database-group)
[https://ravendb.net/docs/article-page/5.2/csharp/studio/server/debug/advanced/cluster-observer#when-the-observer-notices-that-a-node-is-down-or-unresponsive-it-will](https://ravendb.net/docs/article-page/5.2/csharp/studio/server/debug/advanced/cluster-observer#when-the-observer-notices-that-a-node-is-down-or-unresponsive-it-will)

**Comment by logankp** [comment]
Thanks for the reply. It seems like the CPUs were under heavy load for quite a while. We're going to do some testing next time and see if the subscription did successfully transition to another node.

**Comment by logankp** [comment]
Just got more info. Our app was logging this exception over a span of around 36 hours, nothing was happening that would cause load overnight. Re-deploying our app made the exceptions stop

**Comment by ayende** [comment]
Anything in the RavenDB logs?

On Tue, Nov 9, 2021 at 8:49 PM logankp ***@***.***> wrote:

> Just got more info. Our app was logging this exception over a span of
> around 36 hours, nothing was happening that would cause load overnight.
> Re-deploying our app made the exceptions stop
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ravendb/ravendb/discussions/13018#discussioncomment-1614740>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA4RM5H5ICZZKLTWEIVDPLULFUKPANCNFSM5HVS6HIA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
<https://ravendb.net/> *Oren Eini*
*CEO   /   Hibernating Rhinos LTD <https://hibernatingrhinos.com/>*
Mobile:  972-52-548-6969
Sales:  ***@***.***
Skype:  ayenderahien
Support:  ***@***.***
<https://www.facebook.com/pages/RavenDB/265907650186374>
<https://twitter.com/ravendb>
<https://www.linkedin.com/company/hibernating-rhinos-ltd-/>
<https://ravendb.net/emailsignature/displayeventpage>

**Reply by logankp** [reply]
I've been told we can't access any logs outside of the studio session. Are those logs something we can get using RavenDb cloud?

**Reply by garayx** [reply]
Are you a [RavenDB Cloud](https://cloud.ravendb.net/) customer ? Then please open an support ticket using the [portal](https://cloud.ravendb.net/portal/).
